app/validators/account_reset/granted_token_validator.rb

```diff
@@ -0,0 +1,38 @@
+module AccountReset
+  module GrantedTokenValidator
+    extend ActiveSupport::Concern
+
+    included do
+      validates :token, presence: { message: I18n.t('errors.account_reset.granted_token_missing') }
+      validate :token_exists, if: :token_present?
+      validate :token_not_expired, if: :token_present?
+    end
+
+    private
+
+    attr_reader :token
+
+    def token_exists
+      return if request
```

Comment:
Could we use something more meaningful than 'request' so it's not confused with a controller/request?